Added C2 Attribution Commands

RamboV opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

ShacharKidor commented:
Hi @RamboV , we appreciate your effort on contributing to the XSOAR marketplace. Due to excessive amount of contributions in our queue, it may take us a bit more time than usual to review this pull request, we appreciate your patience.

In the meanwhile:
Please fill in the contribution form -
> Thank you for your contribution. Your generosity and caring are unrivaled! Make sure to register your contribution by filling the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form, so our content wizard @ShacharKidor will know he can start review the proposed changes.

ShacharKidor commented on Packs/HYASInsight/Integrations/HYASInsight/README.md:
````diff
@@ -291,128 +299,10 @@ Returns WHOIS records for the provided indicator value.
 #### Command Example
 ```!hyas-get-whois-records-by-indicator indicator_type="domain" indicator_value="domain.net" limit="3"```
````
I'm not sure why the context example and Human Readable Output of 'hyas-get-whois-records-by-indicator' command was removed. was it on purpose?

ShacharKidor commented on Packs/HYASInsight/Integrations/HYASInsight/README.md:
```diff
@@ -600,7 +490,7 @@ Returns associated IP's for the provided hash value.
 | --- | --- | --- |
 | HYAS.HASH-IP.md5 | String | The provided MD5 value | 
-| HYAS.HASH-IP.ips | String | Associated IPS  for the provided MD5 value | 
+| HYAS.HASH-IP.ips | Unknown | Associated IPS  for the provided MD5 value | 
```
```suggestion
| HYAS.HASH-IP.ips | Unknown | Associated IPS for the provided MD5 value | 
```

ShacharKidor commented on Packs/HYASInsight/Integrations/HYASInsight/README.md:
```diff
@@ -623,9 +511,7 @@ Returns associated IP's for the provided hash value.
->|Associated IPs|
->|---|
->| 106.187.43.98 |
+>**No entries.**
```
Would be better to provide an example with existing Human Readable Output entries.

ShacharKidor commented on Packs/HYASInsight/Integrations/HYASInsight/README.md:
```diff
@@ -673,9 +555,101 @@ Returns associated Domain's for the provided hash value.
->| domain.es |
->| qwertasdfg.sinip.es |
->| butterfly.bigmoney.biz |
+>**No entries.**
```
Would be better to provide an example with existing Human Readable Output entries.

ShacharKidor commented on Packs/HYASInsight/ReleaseNotes/1_0_3.md:
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### HYAS Insight
+- Added C2 Attribution Command.
```
```suggestion
- Added the ***hyas-get-c2attribution-records-by-indicator*** Command.
```

ShacharKidor commented on Packs/HYASInsight/ReleaseNotes/1_0_3.md:
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### HYAS Insight
```
I recommend that you add a section that describes the changes you made related to WHOIS commands (URL and endpoint updates).

ShacharKidor commented on Packs/HYASInsight/pack_metadata.json:
```diff
@@ -1,8 +1,8 @@
 {
     "name": "HYAS Insight",
-    "description": "Use the HYAS Insight integration to interactively lookup PassiveDNS, DynamicDNS, WHOIS, Malware Information – either as playbook tasks or through API calls in the War Room.",
+    "description": "Use the HYAS Insight integration to interactively lookup PassiveDNS, DynamicDNS, WHOIS, Malware and C2 Information – either as playbook tasks or through API calls in the War Room.",
```
```suggestion
    "description": "Use the HYAS Insight integration to interactively lookup PassiveDNS, DynamicDNS, WHOIS, Malware and C2 Attribution Information – either as playbook tasks or through API calls in the War Room.",
```

ShacharKidor reviewed (changes requested):
Great Work!
@RamboV - Thank you for contributing. 

In order to complete your contribution: 
1. Please see my minor comments below. 
2. It would be great if you'll add the new command (get-c2attribution-records-by-indicator) to the test-playbook (Playbook-HYASInsight-Test.yml) as well. 

Please let me know if you need any help.

RamboV commented:
> Great Work! @RamboV - Thank you for contributing.
> 
> In order to complete your contribution:
> 
> 1. Please see my minor comments below.
> 2. It would be great if you'll add the new command (get-c2attribution-records-by-indicator) to the test-playbook (Playbook-HYASInsight-Test.yml) as well.
> 
> Please let me know if you need any help.

Hi @ShacharKidor , Thanks for the Feedback.
I have attended to the changes according to your suggestions, please review and provide your feedback.

1. The "get-c2attribution-records-by-indicator" command has already been integrated into the test-playbook.
2. Modified the code change for the limit parameter as suggested in the HyasInsight.py file.
3. Added "outputs_key_field" value as suggested.
4. Added context example and Human Readable Output for 'hyas-get-whois-records-by-indicator'.
5. Removed space for "Associated IPS  for the provided MD5 value" string in the Readme
6. Added context example and Human Readable Output for both "hyas-get-associated-ips-by-hash" and "hyas-get-associated-domains-by-hash" commands
7. Modified "ReleaseNotes/1_0_3.md" as suggested

ShacharKidor commented:
@RamboV - Thank you for your updates. 
One last change - please run the **demisto-sdk fromat** command on the integration's test-playbook you edited, and push the test-playbook again with the changes applied by format command.
For more information please see: [demisto-sdk format](https://xsoar.pan.dev/docs/concepts/demisto-sdk#format)

RamboV commented:
> @RamboV - Thank you for your updates. One last change - please run the **demisto-sdk fromat** command on the integration's test-playbook you edited, and push the test-playbook again with the changes applied by format command. For more information please see: [demisto-sdk format](https://xsoar.pan.dev/docs/concepts/demisto-sdk#format)

Hi @ShacharKidor ,
Ran the command as suggested, please review.
Thanks.

ShacharKidor commented:
> Hi @ShacharKidor ,
> Ran the command as suggested, please review.
> Thanks.

@RamboV - Great! let's schedule a demo. 
I sent you an invitation for a zoom meeting via email, for tomorrow at noon.
Let me know if this is a good time for you.

RamboV commented:
> > Hi @ShacharKidor ,
> > Ran the command as suggested, please review.
> > Thanks.
> 
> @RamboV - Great! let's schedule a demo. I sent you an invitation for a zoom meeting via email, for tomorrow at noon. Let me know if this is a good time for you.

@ShacharKidor Sure that works. Accepted the invitation.
